### PR TITLE
frontend: unify coin and script type into one dropdown

### PR DESCRIPTION
**Pull request by thisconnect**

### CODE REVIEW

File: frontends/web/src/routes/account/add/addaccount.tsx [hunk 1]
```diff
@@ -21,41 +21,103 @@ import { Button, ButtonLink, Input, Select } from '../../../components/forms';
...
+        scriptType: 'p2wpkh',
+    },
+};
+
+
+
 interface State {
-    coinCode: string;
-    scriptType: string;
+    value: string;
```
[the remark this answers is not in the document]
Comment (KasparEtter) [reply frontends/web/src/routes/account/add/addaccount.tsx]: Just googled: `accountType: keyof typeof COIN_AND_ACCOUNT_CODES` could/should work!

File: frontends/web/src/routes/account/add/addaccount.tsx [hunk 2]
```diff
@@ -21,41 +21,103 @@ import { Button, ButtonLink, Input, Select } from '../../../components/forms';
...
+type Props = TestingProps & TranslateProps;
+
+class AddAccount extends Component<Props, State> {
+    constructor(props: Props) {
         super(props);
 
         this.state = {
-            coinCode: 'tbtc',
-            scriptType: 'p2wpkh-p2sh',
+            value: props.testing ? 'tbtc-p2wpkh-p2sh' : 'btc-p2wpkh-p2sh',
```
Comment (KasparEtter) [inline frontends/web/src/routes/account/add/addaccount.tsx]: With `accountType: keyof typeof COIN_AND_ACCOUNT_CODES` above, typos should be detected here. Please try! 🙂

File: frontends/web/src/routes/account/add/addaccount.tsx [hunk 3]
```diff
@@ -21,41 +21,103 @@ import { Button, ButtonLink, Input, Select } from '../../../components/forms';
...
             accountName: '',
             extendedPublicKey: '',
         };
     }
 
     private submit = () => {
-        const body = {
-            coinCode: this.state.coinCode,
-            scriptType: this.state.scriptType,
+        const { coinCode, scriptType } = COIN_AND_ACCOUNT_CODES[this.state.value];
```
Comment (KasparEtter) [inline frontends/web/src/routes/account/add/addaccount.tsx]: I'm not sure whether `coinCode, scriptType` are typed here. Please verify in IDE by hovering over the names. If not, we could/should improve the typing of `COIN_AND_ACCOUNT_CODES`. But maybe typing `this.state.value` as suggested above already helped.

File: frontends/web/src/routes/account/add/addaccount.tsx [hunk 4]
```diff
@@ -21,41 +21,103 @@ import { Button, ButtonLink, Input, Select } from '../../../components/forms';
...
-        };
-        apiPost('account-add', body).then(accountCode => route('/account/' + accountCode));
+        })
+            .then(accountCode => route('/account/' + accountCode));
     }
 
     public render(
-        { t, ...other }: RenderableProps<TranslateProps>,
-        { coinCode, scriptType, accountName, extendedPublicKey }: Readonly<State>,
+        { t, testing, ...other }: RenderableProps<Props>,
```
Comment (KasparEtter) [inline frontends/web/src/routes/account/add/addaccount.tsx]: I don't like these varargs (but you didn't introduce them in your PR).

File: frontends/web/src/routes/account/add/addaccount.tsx (hunk 3 above)
Comment (thisconnect) [reply frontends/web/src/routes/account/add/addaccount.tsx]: Seems as it only knows that this should be a string

**Comment by thisconnect** [comment]
please check again

**Comment by thisconnect** [comment]
force pushed rebased branch

**Pull request merged**